base.mk

```diff
@@ -116,7 +116,7 @@ BUILD ?= release
 CFLAGS += -ffunction-sections -fdata-sections
 LDFLAGS += -Xlinker --gc-sections
 ifeq ($(BUILD), debug)
-CFLAGS += -O0 -g -DDEBUG
+CFLAGS += -Og -g -DDEBUG
```

Comment:
In all my projects code size sooner or later begins to be a problem. As far as I understand -Og is almost -O2 and for other toolchains I've used produces smaller binaries. That is especially important for debug builds where you cannot use more aggressive optimizations and you have to accommodate symbols and other debug related stuff in your binary.